pandas/io/tests/test_pytables.py

```diff
@@ -4733,6 +4733,79 @@ def test_invalid_complib(self):
         with ensure_clean_path(self.path) as path:
             self.assertRaises(ValueError, df.to_hdf, path, 'df', complib='blosc:zlib')
 
+    # GH10447
+    def test_complex(self):
+        df = DataFrame(np.random.rand(4, 5).astype(np.complex64),
+                       index=list('abcd'),
+                       columns=list('ABCDE'))
+        with ensure_clean_path(self.path) as path:
```

Comment:
need tests on query with a complex value. That is the point of all of the kind/typ meta-data.